taskgraph/Task.py

```diff
@@ -52,19 +52,25 @@
 # https://stackoverflow.com/a/8963618/42897
 class NoDaemonProcess(multiprocessing.Process):
     """Make 'daemon' attribute always return False."""
-
-    def _get_daemon(self):
+    @property
+    def daemon(self):
         return False
 
-    def _set_daemon(self, value):
+    @daemon.setter
+    def daemon(self, value):
         pass
-    daemon = property(_get_daemon, _set_daemon)
+
+
+class NoDaemonContext(type(multiprocessing.get_context('spawn'))):

```

Comment:
Just because the `spawn` workaround is such an interesting addition here, could you maybe add a comment about why it's necessary to add this extra context?  I'm a little concerned that future us will look at this in the future and wonder why we added it.

Reply:
@phargogh I think the comments I added should address your comments. Let me know if there should be more elaboration.

Reply:
I think this is just fine ... there's a lot of detail in this text coped from the SO post, so no need to do otherwise, I don't think.